### ISSUE TITLE
Overflow in get-log number of bytes to be read

### ISSUE THREAD

**Issue by den-makarov** [linux-nvme/nvme-cli#656]
Studying source code it was discovered that **nvme_get_log14** function parses cli arguments with assumption that user provides request size in bytes multiple of 4, that is count of DWORDS*4

[nvme-iotcl.c](https://github.com/linux-nvme/nvme-cli/blob/972effdf884eb6d04a996cfe65820698b28975df/nvme-ioctl.c#L420)

If user provides size that it is not multiple of 4, mentioned function truncates two LSB and formats number of DWORDS to be read biased from 0 by subtracting 1

```
__u32 numd = (data_len >> 2) - 1;
__u16 numdu = numd >> 16, numdl = numd & 0xffff;
```

Therefore, if user provides number less than 4, it will turn out to be 0xFFFF FFFF due to integer unsigned wrap around.

Is it OK?

**Comment by keithbusch** [comment]
no, we should validate the user input is always dword sized and aligned for log pages.

**Comment by igaw** [comment]
No progress in this bug report for a long time. If you think it's still relevant please reopen.
